File: Data Structures & Algorithms/remove-element/submission-0.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int k = 0;

        int l = 0 , r = nums.length - 1;

        while (l <= r){
            if (nums[l] == val){
                nums[l] = nums[r];
                r--;
            }else {
                k++;
                l++;
            }
        }

        return k;
    }
}

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    maxSlidingWindow(nums, k) {
        let res = [];
        let l = 0, r = k - 1;

        while (r < nums.length) {
            const subA = nums.slice(l, r + 1);
            res.push(Math.max(...subA))
            l++;
            r++;
        }

        return res
    }
}
